snapcraft/internal/sources/_git.py

```diff
@@ -90,3 +90,25 @@ def pull(self):
             self._pull_existing()
         else:
             self._clone_new()
+        self.assets = self._get_source_details()
+
+    def _get_source_details(self):
```

Comment:
updated to only get the commit if nothing else has been supplied.  I plan for each part to get only what is supplied and to only look up whichever bit is needed to correctly identify the exact source.  Thus I can just directly add 'source-branch', 'source-tag', 'source-commit', etc.  I might have to add some logic to each source class if there are identifiers that are only needed for some source types.